src/interfaces/records/handlers/records-upload.ts

```diff
@@ -0,0 +1,196 @@
+import type { MethodHandler } from '../../types.js';
+import type { DidResolver, MessageStore, UploadStore } from '../../../index.js';
+import type { RecordsUploadCompleteMessage, RecordsUploadPartMessage, RecordsUploadStartMessage } from '../types.js';
+
+import { authenticate } from '../../../core/auth.js';
+import { MessageReply } from '../../../core/message-reply.js';
+import { RecordsUpload } from '../messages/records-upload.js';
+import { DwnInterfaceName, DwnMethodName, DwnStateName } from '../../../core/message.js';
+
+type RecordsUploadMessageVariant = RecordsUploadCompleteMessage | RecordsUploadPartMessage | RecordsUploadStartMessage;
+
+export class RecordsUploadHandler implements MethodHandler {
+
+  constructor(private didResolver: DidResolver, private messageStore: MessageStore, private uploadStore: UploadStore) { }
+
+  public async handle({
```

Comment:
havent written any tests for this yet.  wanted to get the PR up so folks could give feedback on the API :)